src/fileio.c

```diff
@@ -7261,7 +7261,16 @@ delete_recursive(char_u *name)
 # endif
 	    )
     {
-	vim_snprintf((char *)NameBuff, MAXPATHL, "%s/*", name);
+	char_u *esc_name, *esc_name2;
+	/* Escape some regexp chars in file names as these probably will be
+	 * interpreted as ill-formed regular expressions in
+	 * gen_expand_wildcards. */
+	esc_name = do_string_sub(name, (char_u *)"[",
```

Comment:
I don't know this patch is correct.
But, You should do the NULL check to the esc_name and esc_name2.
If NULL, call the vim_free() properly and return -1.